### ISSUE TITLE
TypeErrors when using CacheEtagMiddleware

### ISSUE THREAD

**Issue by rueschm** [Krukov/cashews#190]
when using CacheEtagMiddleware I get errors when the attempt is made to hash the response.

Error is:

```
    etag = blake2s(data).hexdigest()
           │       └ <starlette.responses.Response object at 0x1284100a0>
           └ <class '_blake2.blake2s'>

TypeError: object supporting the buffer API required
```

if used with a Fastapi response as follows:
```
@router.get("/test")
@cache2(ttl="2m")
async def get_test(request: Request, response: Response):
    return Response(content="d")
```

or if used e.g. with a string as response, the error is:

```
    etag = blake2s(data).hexdigest()
           │       └ 'dd'
           └ <class '_blake2.blake2s'>

TypeError: Strings must be encoded before hashing
```

with this code:

```
@router.get("/test")
@cache2(ttl="2m")
async def get_test(request: Request, response: Response):
    return "dd"
```

The functionality works if I use an encoded string as response like:

```
@router.get("/test")
@cache2(ttl="2m")
async def get_test(request: Request, response: Response):
    return "dd".encode()
```

this is also how it is done in your test case ;)

thanks for the amazing work!

**Comment by Krukov** [comment]
Hello, thanks for the reporting an issue and sorry for the wait 

Probably a bug, this functionality not well tested. I will take a look  soon

**Comment by Krukov** [comment]
If so I suggest to you use a quick fix :

```python
cache.setup("mem://", pickle_type="default")

```
 inmemory cache stored values "as is",   but with pickle_type="default" will serialize to pickle ( bytes )

**Comment by Krukov** [comment]
Fixed in 7.0.0
